src/plugins/foxPage/hooks/useCurrentBlockNumber.ts

```diff
@@ -2,26 +2,19 @@ import { useEffect, useState } from 'react'
 
 import { getEthersProvider } from '../utils'
 
+const web3Provider = getEthersProvider()
+
 export const useCurrentBlockNumber = () => {
   const [block, setBlock] = useState<number | null>(null)
 
-  // attach/detach listeners
   useEffect(() => {
-    const web3Provider = getEthersProvider()
-    if (!web3Provider) return
-
-    setBlock(null)
+    if (!web3Provider || block) return
```

Comment:
Note: This is for diff purposes, the next PR will ditch this hook entirely, since it doesn't need to be a hook anymore and we can effectively use `web3Provider.getBlockNumber()` directly